aether-ui/aether/ui/assets/apps/pipeline/redux.jsx

```diff
@@ -136,6 +136,35 @@ const parsePipeline = (pipeline) => {
 }
 
 const reducer = (state = INITIAL_PIPELINE, action) => {
+  const flattenPipelineContracts = pipelines => {
+    let results = []
+    if (pipelines && pipelines.length) {
+      results = pipelines.map(row => (pipelineTranslator(row)))
+    }
+    return results
+  }
+  const pipelineTranslator = pipeline => {
+    const firstContract = pipeline.contracts.length && pipeline.contracts[0]
+    if (firstContract !== null) {
+      pipeline.entity_types = firstContract.entity_types
+      pipeline.mapping = firstContract.mapping
+      pipeline.mapping_errors = firstContract.mapping_errors
+      pipeline.output = firstContract.output
+      pipeline.kernel_refs = firstContract.kernel_refs || {}
+      pipeline.published_on = firstContract.published_on || null
+      pipeline.contract_id = firstContract.id
+    } else {
+      pipeline.entity_types = []
+      pipeline.mapping = []
+      pipeline.mapping_errors = []
+      pipeline.output = {}
+      pipeline.kernel_refs = {}
+      pipeline.published_on = null
+      pipeline.contract_id = null
+    }
+    delete pipeline.contracts
+    return pipeline
+  }
```

Comment:
I can imagine that this (take only the first Contract and not all) will be fixed in further PRs, right?

Reply:
Very correct, this is the "intermediate" mapping set solution as discussed to work with the old UI